src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -432,92 +440,124 @@ gsl::span<const winrt::com_ptr<Profile>> SettingsLoader::_getNonUserOriginProfil
         settings.baseLayerProfile->Origin(OriginTag::ProfilesDefaults);
     }
 
-    // profiles.list
     {
-        const auto size = profilesArray.size();
-
-        // NOTE: This function is supposed to *replace* the contents of ParsedSettings. Don't break this promise.
-        // SettingsLoader::FindFragmentsAndMergeIntoUserSettings relies on this.
-        settings.profiles.clear();
+        const auto size = json.profilesList.size();
         settings.profiles.reserve(size);
-
-        settings.profilesByGuid.clear();
         settings.profilesByGuid.reserve(size);
 
-        for (const auto& profileJson : profilesArray)
+        for (const auto& profileJson : json.profilesList)
         {
-            auto profile = Profile::FromJson(profileJson);
-            profile->Origin(origin);
-
-            // The Guid() generation below depends on the value of Source().
-            // --> Provide one if we got one.
-            if (!source.empty())
+            auto profile = _parseProfile(origin, source, profileJson);
+            if (profile->HasGuid())
             {
-                profile->Source(source);
+                _appendProfile(std::move(profile), settings);
             }
+        }
+    }
+}
+
+// Just like _parse, but is to be used for fragment files, which don't support anything but color
+// schemes and profiles. Additionally this function supports profiles which specify an "updates" key.
+void SettingsLoader::_parseFragment(const winrt::hstring& source, const std::string_view& content, ParsedSettings& settings)
+{
+    const auto json = _parseJson(content);
+
+    settings.clear();
 
-            // The Guid() getter generates one from Name() and Source() if none exists otherwise.
-            // We want to ensure that every profile has a GUID no matter what, not just to
-            // cache the value, but also to make them consistently identifiable later on.
-            if (!profile->HasGuid())
+    {
+        settings.globals = winrt::make_self<GlobalAppSettings>();
+
+        for (const auto& schemeJson : json.colorSchemes)
+        {
+            try
             {
-                if (profile->HasName())
+                if (const auto scheme = ColorScheme::FromJson(schemeJson))
                 {
-                    profile->Guid(profile->Guid());
+                    settings.globals->AddColorScheme(*scheme);
                 }
-                else if (!updatesKeyAllowed || profile->Updates() == winrt::guid{})
+            }
+            CATCH_LOG()
+        }
+    }
+
+    {
+        const auto size = json.profilesList.size();
+        settings.profiles.reserve(size);
+        settings.profilesByGuid.reserve(size);
+
+        for (const auto& profileJson : json.profilesList)
+        {
+            try
+            {
+                auto profile = _parseProfile(OriginTag::Fragment, source, profileJson);
+                if (profile->HasGuid() || profile->Updates() != winrt::guid{})

```

Comment:
similar question as above (though this one might be ok).

Won't we always have a guid, so we'll always append the profile?